### ISSUE TITLE
add a regressor grid to grid_search_forecaster

### ISSUE THREAD

**Issue by tkaraouzene** [JoaquinAmatRodrigo/skforecast#182]
It would be nice to be able to test different regressors using ``grid_search_forecaster``.
It could be done by adding a to possibilities to ``param_grid`` to be a dict.

In this case, each key would be an instanciated regressor, and values its assiciated param grid search, e.g.:

```python
from sklearn.ensemble import RandomForestRegressor, GradientBoostingRegressor

param_grid = {
    RandomForestRegressor(): {'n_estimators': [50, 100], 'max_depth': [5, 10, 15]},
    GradientBoostingRegressor: {'n_estimators': [50, 100], 'max_depth': [5, 10, 15]}
}
```

**Comment by JavierEscobarOrtiz** [comment]
Hello @tkaraouzene,

Thanks for the idea. we will take it into account to implement it in future versions. 😄 

For the moment, this code would allow doing what you say:

```python
# Libraries
# ==============================================================================
import pandas as pd

from sklearn.ensemble import RandomForestRegressor
from sklearn.ensemble import GradientBoostingRegressor
from skforecast.ForecasterAutoreg import ForecasterAutoreg
from skforecast.model_selection import grid_search_forecaster
from sklearn.metrics import mean_squared_error


# Serie
np.random.seed(123)
y = pd.Series(np.random.rand(50), name='y')

# Models to loop
models = [RandomForestRegressor(random_state=123), 
          GradientBoostingRegressor(random_state=123)]

# param_grid to loop for each model
param_grids = {'RandomForestRegressor': {'n_estimators': [50, 100], 'max_depth': [5, 15]},
               'GradientBoostingRegressor': {'n_estimators': [20, 50], 'max_depth': [5, 10]}}

df_results = pd.DataFrame()

for i, model in enumerate(models):

    forecaster = ForecasterAutoreg(
                    regressor = model,
                    lags      = 3
                 )

    # Lags used as predictors
    lags_grid = [3, 5]

    # Regressor hyperparameters
    param_grid = param_grids[list(param_grids)[i]]

    results_grid = grid_search_forecaster(
                            forecaster  = forecaster,
                            y           = y,
                            param_grid  = param_grid,
                            lags_grid   = lags_grid,
                            steps       = 3,
                            refit       = True,
                            metric      = 'mean_squared_error',
                            initial_train_size = len(y) - 12,
                            fixed_train_size   = True,
                            return_best = False,
                            verbose     = False
                   )
    
    # Create a column with model name
    results_grid['model'] = list(param_grids)[i]
    
    df_results = pd.concat([df_results, results_grid])
    
df_results.sort_values(by='metric')
```

|    | lags        | params                                 |    metric |   max_depth |   n_estimators | model                     |
|---:|:------------|:---------------------------------------|----------:|------------:|---------------:|:--------------------------|
|  0 | [1 2 3]     | {'max_depth': 5, 'n_estimators': 20}   | 0.0608184 |           5 |             20 | GradientBoostingRegressor |
|  1 | [1 2 3]     | {'max_depth': 5, 'n_estimators': 50}   | 0.0658502 |           5 |             50 | GradientBoostingRegressor |
|  4 | [1 2 3 4 5] | {'max_depth': 5, 'n_estimators': 20}   | 0.0760274 |           5 |             20 | GradientBoostingRegressor |
|  1 | [1 2 3]     | {'max_depth': 5, 'n_estimators': 100}  | 0.0846158 |           5 |            100 | RandomForestRegressor     |
|  5 | [1 2 3 4 5] | {'max_depth': 5, 'n_estimators': 50}   | 0.0848915 |           5 |             50 | GradientBoostingRegressor |
|  3 | [1 2 3]     | {'max_depth': 15, 'n_estimators': 100} | 0.0849912 |          15 |            100 | RandomForestRegressor     |
|  0 | [1 2 3]     | {'max_depth': 5, 'n_estimators': 50}   | 0.0857305 |           5 |             50 | RandomForestRegressor     |
|  4 | [1 2 3 4 5] | {'max_depth': 5, 'n_estimators': 50}   | 0.0872552 |           5 |             50 | RandomForestRegressor     |
|  7 | [1 2 3 4 5] | {'max_depth': 15, 'n_estimators': 100} | 0.0872771 |          15 |            100 | RandomForestRegressor     |
|  5 | [1 2 3 4 5] | {'max_depth': 5, 'n_estimators': 100}  | 0.0874998 |           5 |            100 | RandomForestRegressor     |
|  2 | [1 2 3]     | {'max_depth': 15, 'n_estimators': 50}  | 0.0875966 |          15 |             50 | RandomForestRegressor     |
|  6 | [1 2 3 4 5] | {'max_depth': 15, 'n_estimators': 50}  | 0.0899773 |          15 |             50 | RandomForestRegressor     |
|  2 | [1 2 3]     | {'max_depth': 10, 'n_estimators': 20}  | 0.0918284 |          10 |             20 | GradientBoostingRegressor |
|  3 | [1 2 3]     | {'max_depth': 10, 'n_estimators': 50}  | 0.0968937 |          10 |             50 | GradientBoostingRegressor |
|  6 | [1 2 3 4 5] | {'max_depth': 10, 'n_estimators': 20}  | 0.0994515 |          10 |             20 | GradientBoostingRegressor |
|  7 | [1 2 3 4 5] | {'max_depth': 10, 'n_estimators': 50}  | 0.107032  |          10 |             50 | GradientBoostingRegressor |

**Comment by JoaquinAmatRodrigo** [comment]
It has been more than 60 days since the last activity on this GitHub issue. Since we have not received any updates or progress reports, we will be closing this issue.

If this issue is still relevant and requires attention, please feel free to reopen it and provide an update. We appreciate your contributions and would love to see this issue resolved if it is still relevant.

Thank you for your participation and cooperation!

**Comment by astrojuanlu** [comment]
Unsure if this was an automated process @JoaquinAmatRodrigo, but wondering if auto-closing should be left only for old bug reports in which the user didn't provide enough context to reproduce? For example other projects use a `needs-more-info` label for those.
